src/core/reactor.cc

```diff
@@ -663,7 +663,12 @@ reactor::signals::signal_handler::signal_handler(int signo, noncopyable_function
 void
 reactor::signals::handle_signal(int signo, noncopyable_function<void ()>&& handler) {
     signal_handler h(signo, std::move(handler));
-    _signal_handlers.insert_or_assign(signo, std::move(h));
+    auto [_, inserted] =  _signal_handlers.insert_or_assign(signo, std::move(h));
+    if (!inserted) {
```

Comment:
Is there any reason this comes as a separate patch? Once emplace is replaced with insert_or_assign it makes no sense _not_ to check for if (!inserted)

Reply:
> Is there any reason this comes as a separate patch?

The `if (!inserted)` check, which is just a small optimization to skip the `sigaction(2)` call, is not this patch mainly aims for, but the replacement of `emplace` by `inserted_or_assign` to allow registering handler multiple times is.

Should I separate these two commits into two pull requests?

> Once emplace is replaced with insert_or_assign it makes no sense _not_ to check for if (!inserted)

Sorry, I don't quiet understand this. Do you mean that we **should not** check `if (!inserted)`  if `emplaced` is still used(which is exactly what the original code did)?

Reply:
> Do you mean that we should not check if (!inserted) if emplaced is still used(which is exactly what the original code did)?

I meant, that the first patch replaces emplace with insert_or_assign, and once this is done the `if (!inserted)` check _must_ be there instantly. There's no point (from my point of view) in adding this check in another patch.

Reply:
I see.

But the original code did not do that check either. And if I **only** replace `emplace` with `insert_or_assign`, the code works well, too. So to keep the modification small, I split it into two patches.